client/dashboard/sites/deployments/index.tsx

```diff
@@ -1,8 +1,32 @@
+import { useQuery } from '@tanstack/react-query';
 import { __ } from '@wordpress/i18n';
+import { siteQuery } from '../../app/queries';
+import { siteRoute } from '../../app/router';
 import PageLayout from '../../components/page-layout';
+import { SitePlan } from '../../data/types';
+import { DeploymentsCallOut } from './deployments-call-out';
 
 function SiteDeployments() {
-	return <PageLayout title={ __( 'Deployments' ) } />;
+	const { siteSlug } = siteRoute.useParams();
+	const { data } = useQuery( siteQuery( siteSlug ) );
+
+	if ( ! data ) {
+		return;
+	}
+
+	const showCallOut =
+		( ! data.site.plan || ! hasDeploymentsFeature( data.site.plan ) ) &&
+		// Show to all atomic sites regardless of plan status e.g. an atomic site in the revert queue
+		// after a plan downgrade will continue to see Deployments until the revert is complete
+		! data.site.options?.is_wpcom_atomic;
+
+	return (
+		<PageLayout title={ __( 'Deployments' ) } callOut={ showCallOut && <DeploymentsCallOut /> } />
+	);
```

Comment:
With how we're using the call-out as a modal-like element not part of the layout's `VStack`, could we render it adjacent to the page layout rather than inside it? To limit prop overload on `PageLayout`.

e.g.

```suggestion
	return (
	    <>
		  <PageLayout title={ __( 'Deployments' ) } />
		  { showCallOut && <DeploymentsCallOut /> }
		</>
	);
```

Reply:
From an a11y perspective I suppose the Callout will need to be outside because `PageLayout` contents will need to be `inert` or `aria-hidden="true"`.

I guess it will also need to trap focus inside the 'modal' too, but still provide access back to the main nav.

Alternatively we could just use an actual modal dialog. I suspect this would be much simpler 😅

Reply:
I'm actually not sure we'll have any content outside the callout, most endpoints ... will fail unless the user actually has the right plans... so the content outside the callout probably needs to be "fake". We might need a system for that.

Reply:
That's a good point, I prefer this.
The downside is the call-out will no longer have access to the page layout's height, which it can use to ensure all the components on the page get covered, without resorting to `position: fixed`.